tariff/ngeso/api.go

```diff
@@ -0,0 +1,139 @@
+const BaseURI = "https://api.carbonintensity.org.uk/"
+
+// ForecastNationalURI defines the location of the national forecast.
+// Replace the first %s with the RFC3339 timestamp to fetch from.
+const ForecastNationalURI = BaseURI + "intensity/%s/fw48h"
+
+// ForecastRegionalByIdURI defines the location of the regional forecast determined by Region ID.
+// Replace the first %s with the RFC3339 timestamp to fetch from, and the second with the appropriate Region ID.
+const ForecastRegionalByIdURI = BaseURI + "regional/intensity/%s/fw48h/regionid/%s"
+
+// ForecastRegionalByPostcodeURI defines the location of the regional forecast determined by a given postcode.
+// Replace the first %s with the RFC3339 timestamp to fetch from, and the second with the appropriate postcode.
+const ForecastRegionalByPostcodeURI = BaseURI + "regional/intensity/%s/fw48h/postcode/%s"
+
+// ConstructNationalForecastRequest returns a request object to be used when calling the national API.
+func ConstructNationalForecastRequest() *CarbonForecastNationalRequest {
+	return &CarbonForecastNationalRequest{}
+}
+
+// ConstructRegionalForecastByIDRequest returns a validly formatted, fully qualified URI to the forecast valid for the given region.
+func ConstructRegionalForecastByIDRequest(r string) *CarbonForecastRegionalRequest {
+	return &CarbonForecastRegionalRequest{regionid: r}
+}
+
+// ConstructRegionalForecastByPostcodeRequest returns a validly formatted, fully qualified URI to the forecast valid for the given postcode.
+func ConstructRegionalForecastByPostcodeRequest(p string) *CarbonForecastRegionalRequest {
+	return &CarbonForecastRegionalRequest{postcode: p}
+}
+
+type CarbonForecastRequest interface {
+	URI() (string, error)
+	DoRequest(helper *request.Helper) (CarbonForecastResponse, error)
+}
+
+type CarbonForecastNationalRequest struct{}
+
+func (r *CarbonForecastNationalRequest) URI() (string, error) {
+	currentTs := time.Now().UTC()
+	t := currentTs.Format(time.RFC3339)
+	return fmt.Sprintf(ForecastNationalURI, t), nil
+}
+
+func (r *CarbonForecastNationalRequest) DoRequest(client *request.Helper) (res CarbonForecastResponse, err error) {
+	uri, err := r.URI()
+	if err != nil {
+		return
+	}
+	res = &NationalIntensityResult{}
+	err = client.GetJSON(uri, &res)
+	return
+}
+
+type CarbonForecastRegionalRequest struct {
+	regionid string
+	postcode string
+}
+
+func (r *CarbonForecastRegionalRequest) URI() (string, error) {
+	currentTs := time.Now().UTC()
+	t := currentTs.Format(time.RFC3339)
+	if r.regionid != "" {
+		return fmt.Sprintf(ForecastRegionalByIdURI, t, r.regionid), nil
+	} else if r.postcode != "" {
+		return fmt.Sprintf(ForecastRegionalByPostcodeURI, t, r.postcode), nil
+	}
+
+	return "", ErrRegionalRequestInvalidFormat
+}
+
+func (r *CarbonForecastRegionalRequest) DoRequest(client *request.Helper) (res CarbonForecastResponse, err error) {
+	uri, err := r.URI()
+	if err != nil {
+		return
+	}
+	res = &RegionalIntensityResult{}
```

Comment:
var res …

Reply:
Seems you could also remove this method- avoids the additional dependency on the helper here if you ever want to use this package somewhere else..